### PR TITLE
Ignore dynamic objects from the sharedSchema

### LINKED ISSUE #1783: Don't make a table for RLMDynamicObject / DynamicObject / MigrationObject

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1783.

\c @jpsim  @tgoyne

### CODE REVIEW

File: RealmSwift/Object.swift [hunk 1]
```diff
@@ -260,6 +260,10 @@ public final class DynamicObject : Object {
     public override func setValue(value: AnyObject?, forUndefinedKey key: String) {
         self[key] = value
     }
+
+    @objc private class func shouldPersistToRealm() -> Bool {
+        return false;
```
Comment (tgoyne) [inline RealmSwift/Object.swift]: This looks blatantly wrong. Is it somehow not?

Comment (segiddins) [inline RealmSwift/Object.swift]: Why would we ever want to save DynamicObject?

-Samuel E. Giddins

On Apr 23, 2015, at 5:32 PM, Thomas Goyne <notifications@github.com> wrote:

In RealmSwift/Object.swift:

> @@ -260,6 +260,10 @@ public final class DynamicObject : Object {
>      public override func setValue(value: AnyObject?, forUndefinedKey key: String) {
>          self[key] = value
>      }
> +
> +    @objc private class func shouldPersistToRealm() -> Bool {
> +        return false;
This looks blatantly wrong. Is it somehow not?

Comment (tgoyne) [inline RealmSwift/Object.swift]: Oh, I missed that it was in DynamicObject and not Object...

**Comment by jpsim** [comment]
Since we don't specify Realm for any other schema-related methods (`ignoredProperties`/`primaryKey`/`defaultPropertyValues`), I'd prefer if this one was just called `shouldPersist`. Otherwise :+1:

**Comment by tgoyne** [comment]
:+1:

**Comment by segiddins** [comment]
@jpsim the reason I made this method more verbose is that it's not exposed publicly and I wanted to limit the chances of a name collision.

**Comment by jpsim** [comment]
:+1:

**Pull request merged**